metrics/cli/docs/features/doctor_output_improvements/diagrams/validation_class_diagram.puml

```diff
@@ -0,0 +1,64 @@
+@startuml validation_output_class_diagram
+
+package validation.lib.src {
+    package model {
+        package builder {
+            class ValidationResultBuilder {
+                - _results: Map<ValidationTarget, TargetValidationResult>
+
+                + setResult(result: TargetValidationResult): void
+                + setEmptyResults(result: TargetValidationResult): void
+                + build(): ValidationResult
+            }
```

Comment:
Should we display the `createForTargets` factory method in this class? Also, I believe it would be great to mention that this class has a private constructor.